Lib/collections/__init__.py

```diff
@@ -788,7 +788,12 @@ def __eq__(self, other):
         'True if all counts agree. Missing counts are treated as zero.'
         if not isinstance(other, Counter):
             return NotImplemented
-        return all(self[e] == other[e] for c in (self, other) for e in c)
+
+        for obj1, obj2 in ((self, other), (other, self)):
+            for k, v in obj1.items():
+                if v != obj2[k]:
+                    return False
```

Comment:
The is faster because it replaces half the (Python) indexings with (C) iteration through obj1.items.  I believe the following is equivalent:
```
            if any(v != obj2[k] for k, v in obj1.items())
                return False
```
If so, how does speed compare?